pandas/tseries/plotting.py

```diff
@@ -278,8 +280,22 @@ def _maybe_convert_index(ax, data):
 # Patch methods for subplot. Only format_dateaxis is currently used.
 # Do we need the rest for convenience?
 
-
-def format_dateaxis(subplot, freq):
+def timeTicks(x, pos, n_decimals):
+    ''' Convert seconds to 'D days HH:MM:SS.F' '''
+    s, ns = divmod(x, 1e9)
+    m, s = divmod(s, 60)
+    h, m = divmod(m, 60)
+    d, h = divmod(h, 24)
+    decimals = int(ns * 10**(n_decimals - 9))
+    s = r'{:02d}:{:02d}:{:02d}'.format(int(h), int(m), int(s))
+    if n_decimals > 0:
```

Comment:
of course it does. This is pandas Timedelta. (you are looking at the printing), which by-default suppres it

```
In [6]: pd.Timedelta(1, unit='ns')._repr_base(format='all')
Out[6]: '0 days 00:00:00.000000001'
```